### PR TITLE
Improve readme to get started with plotly.js JSON interface for non-JavaScript developers and adjust dist/README

### PR DESCRIPTION
**Pull request by archmoj**
@plotly/plotly_js

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,21 +18,58 @@ Plotly.js can be used to produce dozens of chart types and visualizations, inclu
...
+While non-minified source files may contain characters outside UTF-8, it is recommended that you specify the `charset` when loading those bundles.
+```html
+    <script src="https://cdn.plot.ly/plotly-2.0.0-rc.2.js" charset="utf-8"></script>
+```
+
+#### Please note that as of v2 the "plotly-latest" outputs (e.g. https://cdn.plot.ly/plotly-latest.min.js) will no longer be updated on the CDN, and will stay at the last v1 patch v1.58.4. Therefore, to use the CDN with plotly.js v2 and higher, you must specify an exact plotly.js version.
 
-### Install with npm
+
+## Load from npm
```
Comment (nicolaskruchten) [inline README.md:72]: I would have this be first

File: README.md [hunk 2]
```diff
@@ -18,21 +18,58 @@ Plotly.js can be used to produce dozens of chart types and visualizations, inclu
...
 * [Documentation](#documentation)
 * [Contributing](#contributing)
 * [Community](#community)
-* [Versioning](#versioning)
 * [Notable contributors](#creators)
 * [Copyright and license](#copyright-and-license)
 
 ---
-## Quick start options
+## Load from CDN
```
Comment (nicolaskruchten) [inline README.md:34]: I would spell out "Content Delivery Network (CDN)"

File: README.md [hunk 3]
```diff
@@ -18,21 +18,58 @@ Plotly.js can be used to produce dozens of chart types and visualizations, inclu
...
+Fastly supports Plotly.js with free CDN service. Read more at <https://www.fastly.com/open-source>.
+
+### Minified plotly.js X.Y.Z bundles
```
Comment (nicolaskruchten) [inline README.md:37]: I would rename this "Usage example"

File: README.md [hunk 4]
```diff
@@ -18,21 +18,58 @@ Plotly.js can be used to produce dozens of chart types and visualizations, inclu
...
+As demonstrated in the example above basic knowledge of `html` and [JSON](https://en.wikipedia.org/wiki/JSON) syntax is enough to get started i.e. with/without JavaScript!
+To learn and build more with plotly.js please visit [plotly.js documentation](https://plotly.com/javascript).
+
+### Un-minified versions are also available on CDN
+While non-minified source files may contain characters outside UTF-8, it is recommended that you specify the `charset` when loading those bundles.
+```html
+    <script src="https://cdn.plot.ly/plotly-2.0.0-rc.2.js" charset="utf-8"></script>
+```
+
+#### Please note that as of v2 the "plotly-latest" outputs (e.g. https://cdn.plot.ly/plotly-latest.min.js) will no longer be updated on the CDN, and will stay at the last v1 patch v1.58.4. Therefore, to use the CDN with plotly.js v2 and higher, you must specify an exact plotly.js version.
```
Comment (nicolaskruchten) [inline README.md:69]: for little "notes" like this, please don't use `####` but rather prefix them with `>` so they get indented/dimmed a bit.

File: README.md [hunk 5]
```diff
@@ -42,37 +79,19 @@ and import plotly.js as
...
-
-##### Please note that as of v2 the "plotly-latest" outputs (e.g. https://cdn.plot.ly/plotly-latest.min.js) will no longer be updated on the CDN, and will stay at the last v1 patch v1.58.4. Therefore, to use the CDN with plotly.js v2 and higher, you must specify an exact plotly.js version.
-
-Fastly supports Plotly.js with free CDN service. Read more at <https://www.fastly.com/open-source>
-
-### Download the latest release or a release candidate (rc)
-
-[Latest and rc releases on GitHub](https://github.com/plotly/plotly.js/releases/)
+---
+## Versioning
```
Comment (nicolaskruchten) [inline README.md:89]: This probably doesn't need to be a top-level item, just a little note somewhere, maybe under the NPM section. it's kind of the norm anyway.

File: README.md (hunk 1 above)
Comment (archmoj) [reply README.md]: Done in https://github.com/plotly/plotly.js/pull/5705/commits/a0bd25835c3f415dece9c5cd8604663a60d7f532.

File: README.md (hunk 2 above)
Comment (archmoj) [reply README.md]: Done in https://github.com/plotly/plotly.js/pull/5705/commits/a0bd25835c3f415dece9c5cd8604663a60d7f532.

File: README.md (hunk 3 above)
Comment (archmoj) [reply README.md]: Done in https://github.com/plotly/plotly.js/pull/5705/commits/a0bd25835c3f415dece9c5cd8604663a60d7f532.

File: README.md (hunk 4 above)
Comment (archmoj) [reply README.md]: Done in https://github.com/plotly/plotly.js/pull/5705/commits/a0bd25835c3f415dece9c5cd8604663a60d7f532.

File: README.md (hunk 5 above)
Comment (archmoj) [reply README.md]: Done in https://github.com/plotly/plotly.js/pull/5705/commits/a0bd25835c3f415dece9c5cd8604663a60d7f532.

**Comment by nicolaskruchten** [comment]
Looks better, but why is the MathJax section under Documentation?

**Comment by nicolaskruchten** [comment]
I'd put it in the CDN section I think

File: README.md [hunk 6]
```diff
@@ -84,6 +84,13 @@ While non-minified source files may contain characters outside UTF-8, it is reco
...
+### To support MathJax
+Load relevant MathJax (v2) files *Before* the plotly.js script tag:
+```html
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_SVG.js"></script>
+    <script src="https://cdn.plot.ly/plotly-2.0.0-rc.2.min.js"></script>
+```
+
 ## Partial bundles
 
 There are two kinds of plotly.js partial bundles:
```
Comment (nicolaskruchten) [inline README.md:96]: could we maybe define what a partial bundle is and why we talk about them?

File: README.md [hunk 7]
```diff
@@ -91,10 +91,9 @@ Load relevant MathJax (v2) files *Before* the plotly.js script tag:
...
 ```
 
-## Partial bundles
-
-There are two kinds of plotly.js partial bundles:
-1. The official partial bundles that are distributed to `npm` and the CDN, described in [the dist README](https://github.com/plotly/plotly.js/blob/master/dist/README.md).
+## Bundles
+There are two kinds of plotly.js bundles:
+1. Complete and partial official bundles that are distributed to `npm` and the `CDN`, described in [the dist README](https://github.com/plotly/plotly.js/blob/master/dist/README.md).
 2. Custom bundles you can create yourself, if none of the distributed packages meet your needs.
```
Comment (nicolaskruchten) [inline README.md:97]: I think this custom-bundles section could live in its own separate file, which would be linked to from here (and then all three of dist/custom-bundles/building readmes would link to each other as they are essentially alternative ways of doing similar things)

**Comment by nicolaskruchten** [comment]
The BUILDING.md file doesn't really provide a decent motivation for why you'd want to build Plotly.js into your library or explain the modules concept etc... Can we take some of what used to be in "modules" in the main readme and transplant it into Building please? Basically show folks how they can directly import parts of Plotly into their app (a) and then *if* they import modules that require special webpack/"ify"-handling, tell them how to do it? 

Because if someone wants to build their own app, with just the bar-chart bundle imported directly, they don't need all the "ify" stuff, right?

**Comment by archmoj** [comment]
OK. There are still things that we could improve with the READMEs.
In the past few commits I improved `tasks/stats.js` which writes `dist/README.md`.
Hope it reads better now.
@alexcjohnson I'd appreciate a quick review and I'd be happy to work on the remainings in a separate PR : )

**Comment by alexcjohnson** [comment]
This all looks good to me, but you've got @nicolaskruchten's comments here, wouldn't it be easier to just address the last couple in this PR and be done with it?

**Comment by archmoj** [comment]
> This all looks good to me, but you've got @nicolaskruchten's comments here, wouldn't it be easier to just address the last couple in this PR and be done with it?

The diff is already big and we want to move more lines to new files.
So I'd rather having a separate PR to better address/review those comments.

**Comment by archmoj** [comment]
new `CUSTOM_BUNDLE` file added in https://github.com/plotly/plotly.js/pull/5705/commits/d6b5498ba74636fea93a2da813413acdb270618f

**Comment by nicolaskruchten** [comment]
💃

**Pull request merged**